pdudaemon/drivers/devantech.py

```diff
@@ -61,8 +61,10 @@ def port_interaction(self, command, port_number):
         if port_number > self.port_count:
             log.error("There are only %d ports. Provide a port number lesser than %d." % (self.port_count, self.port_count))
             raise RuntimeError("There are only %d ports. Provide a port number lesser than %d." % (self.port_count, self.port_count))
-
-        if command == "on":
+        if self.logic not in ["NO", "NC"]:
+            log.error("Invalid logic setting: %s." % (self.logic))
+            return
+        elif command == "on":
```

Comment:
Don't use `elif` here, as it's checking something unrelated to the check above. Also, the error case, the function returns before this line is ever executed, making the `elif` redundant.

Reply:
Thanks for the comment, It is converted to if.